### PR TITLE
Dragen umi

### PR DESCRIPTION
**Pull request by skanwal**
**Dragen-UMI pipeline definition**

- Using version 3.8.4
- Inputs include:
  - option to run against fastq inputs - for tumor-only pipeline
  - BAMs - Currently UMI Dragen pipeline does not support tumor-normal mode. 
  To run ctDNA samples in this setting, bam files created separately from tumour-only UMI pipeline and normal sample can be used as an input for doing variant calling.
  - Usual tumor-normal parameters - for tumor-normal run

### CODE REVIEW

File: tools/dragen-umi/3.8.4/dragen-umi__3.8.4.cwl [hunk 1]
```diff
@@ -0,0 +1,665 @@
...
+    inputBinding:
+      prefix: "--umi-min-supporting-reads"
+  umi_enable:
+    label: umi enable
+    doc: |
+      To enable read collapsing, set the --umi-enable option to true. 
+      If using the --umi-library-type option, --umi-enable is not required.
+    type: boolean?
+    inputBinding:
+      prefix: "--umi-enable"
```
[the remark this answers is not in the document]
Comment (alexiswl) [reply tools/dragen-umi/3.8.4/dragen-umi__3.8.4.cwl:323]: Normal doesn't contain umi. Resolved.

**Comment by ohofmann** [comment]
Does DRAGEN support mixed runs (one sample with UMIs, the other without)?

**Comment by skanwal** [comment]
@ohofmann - Currently UMI Dragen pipeline does not support tumor-normal mode. 
So - BAMs for tumor and normal sample are needed to be created in different runs.

**Comment by alexiswl** [comment]
Might be worth adding in a little comment that `bam_input` and `tumor_bam_input` are incompatible with `fastq-file*` inputs.

File: tools/dragen-umi/3.8.4/dragen-umi__3.8.4.cwl [hunk 2]
```diff
@@ -0,0 +1,665 @@
...
+s:author:
+    class: s:Person
+    s:name: Sehrish Kanwal
+    s:email: sehrish.kanwal@umccr.org
+
+# ID/Docs
+id: dragen-umi--3.8.4
+label: dragen-umi v(3.8.4)
+doc: |
+    Documentation for dragen-umi v3.8.4
```
[the remark this answers is not in the document]
Comment (skanwal) [reply tools/dragen-umi/3.8.4/dragen-umi__3.8.4.cwl]: Resolved https://github.com/umccr/cwl-ica/pull/20/commits/cef1dfd3424efcc0f7b59b7f501107015154f111

**Review by alexiswl — APPROVED** [review]

**Comment by alexiswl** [comment]
Catalogue changes completed, ready to merge!

**Pull request merged**